File: 3194-MinimumAverageofSmallestandLargestElements/3194-MinimumAverageofSmallestandLargestElements.cpp
// Last updated: 3/9/2026, 5:24:44 PM
1class Solution {
2public:
3    double minimumAverage(vector<int>& nums) {
4        sort(nums.begin(), nums.end());
5
6        int i = 0;
7        int j = nums.size() - 1;
8        double minAvg = INT_MAX;
9
10        while(i < j){
11            double avg = (nums[i] + nums[j]) / 2.0;
12            minAvg = min(minAvg, avg);
13            i++;
14            j--;
15        }
16
17        return minAvg;
18    }
19};

File: 3194-MinimumAverageofSmallestandLargestElements/3194-MinimumAverageofSmallestandLargestElements.js
// Last updated: 3/9/2026, 5:20:09 PM
1/**
2 * @param {number[]} nums
3 * @return {number}
4 */
5var minimumAverage = function(nums) {
6    nums = nums.sort((a,b)=>a-b);
7
8    let left = 0;
9    let right = nums.length-1;
10    let minAvg = +Infinity;
11
12    while(left < right){
13        let avg = (nums[left]+nums[right]) / 2;
14        minAvg = Math.min(minAvg , avg);
15        left++;
16        right--;
17    }
18    return minAvg
19};